x-pack/plugins/enterprise_search/public/applications/app_search/components/role_mappings/role_mappings_logic.test.ts

```diff
@@ -0,0 +1,437 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { mockFlashMessageHelpers, mockHttpValues, mockKibanaValues } from '../../../__mocks__';
+import { LogicMounter } from '../../../__mocks__/kea.mock';
+
+import { engines } from '../../__mocks__/engines.mock';
+
+import { nextTick } from '@kbn/test/jest';
+
+import { asRoleMapping } from '../../../shared/role_mapping/__mocks__/roles';
+import { ANY_AUTH_PROVIDER } from '../../../shared/role_mapping/constants';
+
+import { RoleMappingsLogic } from './role_mappings_logic';
+
+describe('RoleMappingsLogic', () => {
+  const { http } = mockHttpValues;
+  const { navigateToUrl } = mockKibanaValues;
+  const { clearFlashMessages, flashAPIErrors } = mockFlashMessageHelpers;
+  const { mount } = new LogicMounter(RoleMappingsLogic);
+  const defaultValues = {
+    attributes: [],
+    availableAuthProviders: [],
+    elasticsearchRoles: [],
+    roleMapping: null,
+    roleMappings: [],
+    roleType: 'owner',
+    attributeValue: '',
+    attributeName: 'username',
+    dataLoading: true,
+    hasAdvancedRoles: false,
+    multipleAuthProvidersConfig: false,
+    availableEngines: [],
+    selectedEngines: new Set(),
+    accessAllEngines: true,
+    selectedAuthProviders: [ANY_AUTH_PROVIDER],
+  };
+
+  const mappingsServerProps = { multipleAuthProvidersConfig: true, roleMappings: [asRoleMapping] };
+  const mappingServerProps = {
+    attributes: [],
+    authProviders: [],
+    availableEngines: engines,
+    elasticsearchRoles: [],
+    hasAdvancedRoles: false,
+    multipleAuthProvidersConfig: false,
+    roleMapping: asRoleMapping,
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mount();
+  });
+
+  it('has expected default values', () => {
+    expect(RoleMappingsLogic.values).toEqual(defaultValues);
+  });
+
+  describe('actions', () => {
+    it('setRoleMappingsData', () => {
```

Comment:
Addressed here. I think you meant to nest an `if` instead of a second `describe` in your comment example. Correct me if I am wrong. (https://github.com/elastic/kibana/pull/94461/commits/eb1298a3bd9c43e9251d110e70c290dce6be9966)